### PR TITLE
Improve file system blob loader and generic loader

### PR DESCRIPTION
**Pull request by eyurtsev**
* Add support for passing a specific file to the file system blob loader
* Allow specifying a class parameter for the parser for the generic loader

```python

class AudioLoader(GenericLoader):
  @staticmethod
  def get_parser(**kwargs):
     return MyAudioParser(**kwargs):
```

The intent of the GenericLoader is to provide on-ramps from different sources (e.g., web, s3, file system).

An alternative is to use pipelining syntax or creating a Pipeline

```
FileSystemBlobLoader(...) | MyAudioParser
```

### CODE REVIEW

**Comment by eyurtsev** [comment]
Actually this won't work -- need to make it possible to lazily instantiate -- one sec

File: libs/langchain/langchain/document_loaders/concurrent.py [hunk 1]
```diff
@@ -75,7 +74,11 @@ def from_filesystem(
             show_progress=show_progress,
         )
         if isinstance(parser, str):
-            blob_parser = get_parser(parser)
+            if parser == "default" and cls.get_parser != GenericLoader.get_parser:
```
Comment (efriis) [inline libs/langchain/langchain/document_loaders/concurrent.py:77]: Would it be better to try/except NotImplementedError instead of checking method != ?

**Review by efriis — COMMENTED** [review]
1 question, otherwise lgtm

File: libs/langchain/langchain/document_loaders/concurrent.py (hunk 1 above)
Comment (eyurtsev) [reply libs/langchain/langchain/document_loaders/concurrent.py:77]: Sure that works as well I'll change

**Review by baskaryan — APPROVED** [review]

**Pull request merged**